app/models/user.rb

```diff
@@ -1,17 +1,41 @@
 class User < ApplicationRecord
+  attr_accessor :remember_token
   before_save :email_downcase
   has_secure_password
-
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
-
   validates :email, presence: true, length: {maximum: 255},
     format: {with: VALID_EMAIL_REGEX},
     uniqueness: {case_sensitive: false}
   validates :name, presence: true, length: {maximum: 50}
   validates :password, presence: true, length: {minimum: 6}
 
+  def authenticated?(remember_token)
+    return false unless remember_digest
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
+  end
+
+  def remember
+    self.remember_token = User.new_token
+    update_attributes remember_digest: User.degest(remember_token)
```

Comment:
cho nên là khi mình sử dụng Update_attributes sẽ có thể bảo toàn được tất cả các dữ liệu của object.